lib/puppet/provider/login.rb

```diff
@@ -16,26 +16,59 @@
 
 require 'json'
 
+ONEVIEW_AUTH_FILE = 'ONEVIEW_AUTH_FILE'.freeze
+ONEVIEW_URL = 'ONEVIEW_URL'.freeze
+I3S_AUTH_FILE = 'I3S_AUTH_FILE'.freeze
+I3S_URL = 'I3S_URL'.freeze
+
 # This method returns the information necessary in order to log in to the Oneview Appliance
 # The three possible ways of declaring the variables are, respectively:
 def login
   begin
     # - Creating a JSON file with the proper fields and setting the environment variable ONEVIEW_AUTH_FILE to its path
-    credentials = if ENV['ONEVIEW_AUTH_FILE']
-                    JSON.parse(File.read(File.absolute_path(ENV['ONEVIEW_AUTH_FILE'])), symbolize_names: true)
+    credentials = if ENV[ONEVIEW_AUTH_FILE]
+                    load_credentials_from_file_env_var(ONEVIEW_AUTH_FILE)
                   # - Declaring each field as an environment variable
-                  elsif ENV['ONEVIEW_URL']
+                  elsif ENV[ONEVIEW_URL]
                     environment_credentials
                   # - Placing a JSON file in the directory you are running the manifests from
                   else
-                    JSON.parse(File.read(File.expand_path(Dir.pwd + '/login.json', __FILE__)), symbolize_names: true)
+                    load_credentials_from_file('/login.json')
                   end
   rescue
     raise('The Oneview credentials could not be set. Please check the documentation for more information.')
   end
   credentials_parse(credentials)
 end
 
+# This method returns the information necessary in order to log in to the Image Streamer Appliance
+# The three possible ways of declaring the variables are, respectively:
+def login_i3s
+  begin
+    # - Creating a JSON file with the proper fields and setting the environment variable I3S_AUTH_FILE to its path
+    credentials = if ENV[I3S_AUTH_FILE]
```

Comment:
If we want two different login methods, and their bodies are basically the same, in order to avoid code duplication we would at least want to move the body out into a separate method and make it so the login methods just pass in the differences to that.